src/Renderer/Simulator/CanvasStateAnalyser.fs

```diff
@@ -471,20 +483,63 @@ let checkCustomComponentsOk ((comps, _): CanvasState) (sheets: LoadedComponent l
         | Error(c, BadOutputs(cName, instOuts, compOuts)) ->
             let instOus, compOuts = disp instOuts, disp compOuts
 
-            error c
+            error c OutPortMismatch
             <| sprintf
                 "Sheet %s is used as a custom component. Instance Out ports: %A are different from Component Out ports: %A."
                 cName
                 instOuts
                 compOuts
 
+let getPortNumFromConnPort (connPort: Port) (targetPortList: Port list) =
+    targetPortList
+    |> List.tryFind (fun (port: Port) -> port.Id = connPort.Id)
+    |> function
+        | Some {PortNumber = Some n} -> n
+        | _ -> failwithf "Existing connection should be connected to existing port"
+
+/// Check whether Adders have a NotConnected component connected to their COUT
+/// this is unnecessary since it can be disabled via its options
+let checkAdderUnnecessaryNC ((comps, conns): CanvasState) : SimulationError option =
+    let isIdofNotConnectedComp id =
```

Comment:
isIdofNotConnectedComponent: name. 
1) Camelcase Of if you keep this.
2) Better to make less verbose isNotConnected  (Component implied, id imples, by types)